src/main/scala/nl/knaw/dans/deposit/DepositProperties.scala

```diff
@@ -269,9 +273,12 @@ case class FedoraId(value: Option[String] = None) {
 }
 
 case class Doi(value: Option[String] = None,
-               registered: Option[Boolean] = Some(false)) {
-  def this(doiValue: String, doiRegistered: String) = {
-    this(Option(doiValue), Option(doiRegistered).map(BooleanUtils.toBoolean))
+               registered: Option[Boolean] = Some(false),
+               action: Option[Action] = Some(create)) {
+  def this(doiValue: String, doiRegistered: String, action: String) = {
+    this(Option(doiValue),
+      Option(doiRegistered).map(BooleanUtils.toBoolean),
+      Action.values.find(_.toString == action))
```

Comment:
what happens when someone gives an invalid `action` String? If I see it correctly, it will translate to a `None`. However, in this case we may want to actually use `withName` (as is done in all other contructors) instead, just to make sure that an invalid `action` String results in a user error.

Yes, this isn't proper, as this throws an exception, but I think we'll fix that later by converting this second constructor into a `def apply(doiValue: String, doiRegistered: String, action: String): Try[Doi]`. Maybe you would like to pick that up in a separate PR (once this is merged)?

Reply:
Changed to the with name constructor.

Reply:
Will make a new branch once this one is merged

Reply:
Great! Looking forward to that. 👍